klippy/gcode.py

```diff
@@ -710,3 +710,8 @@ def cmd_HELP(self, params):
             if cmd in self.gcode_help:
                 cmdhelp.append("%-10s: %s" % (cmd, self.gcode_help[cmd]))
         self.respond_info("\n".join(cmdhelp))
+    cmd_RPI_SHUTDOWN_when_not_ready = True
+    cmd_RPI_SHUTDOWN_help = "Turn octoprint and the raspberry off"
+    def cmd_RPI_SHUTDOWN(self, params):
+        os.system("sudo shutdown -h now")
```

Comment:
this also assumes sudo access and that all of these params are available. fairly certain klipper can be ran on more than raspian so you would likely want to ensure the commands are going to work before exposing the menu option or gcode? at the least a try catch and error message?